libs/agent-sm/bootstrap/src/main/java/org/opensearch/javaagent/bootstrap/AccessController.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.javaagent.bootstrap;
+
+import java.util.concurrent.Callable;
+import java.util.function.Supplier;
+
+/**
+ * Utility class to run code in a privileged block.
```

Comment:
Let's expand on this by including some text about when and why this is needed.

Reply:
Expanded on this javadoc with example usages and more explanation. It doesn't quite get into how it relates to the policy files, but perhaps we can link to https://github.com/opensearch-project/project-website/pull/3784 once that's published?